File: PRACTICAS-PDOO/IRRGARTEN/Versiones_en_Java_Ruby/PROYECTO1-JAVA/src/irrgarten/TestP1.java

package irrgarten;

import java.util.Random;

/**
 *
 * @author angel_rodriguez
 */
public class TestP1 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        
        Weapon weapon1 = new Weapon(9, 8);
        Shield shield1 = new Shield(16, 4);
        GameState gameState1 = new GameState("open", "good", "bored", 2, Boolean.TRUE, "GOOO");
        
        // Pruebo la clase Weapon.
        System.out.println("Prueba de la clase Weapon: ");
        System.out.println(weapon1.attack());
        System.out.println(weapon1.toString());
        System.out.println(weapon1.discard());
        
        // Pruebo la clase Shield.
        System.out.println("\nPrueba de la clase Shield: ");
        System.out.println(shield1.protect());
        System.out.println(shield1.toString());
        System.out.println(shield1.discard());

        // Pruebo la clase Dice.
        System.out.println("\nPrueba de la clase Dice: ");     
        System.out.println("randomPos: " + Dice.randomPos(3));
        System.out.println("whoStarts: " + Dice.whoStarts(4));
        System.out.println("randomIntelligence: " + Dice.randomIntelligence());
        System.out.println("randomStrength(): " + Dice.randomStrength());
        System.out.println("resurrectPlayer: " + Dice.resurrectPlayers());
        System.out.println("weaponsReward: " + Dice.weaponsReward());
        System.out.println("shieldsReward: " + Dice.shieldsReward());
        System.out.println("healthReward: " + Dice.healthReward());
        System.out.println("weaponPower: " + Dice.weaponPower());
        System.out.println("shieldPower: " + Dice.shieldPower());
        System.out.println("usesLeft: " + Dice.usesLeft());
        System.out.println("intensity: " + Dice.intensity(2));
        System.out.println("discardElement: " + Dice.discardElement(3));

        // Pruebo la clase GameState.
        System.out.println("\nPrueba de la clase GameState: ");     
        System.out.println(gameState1.getLabyrinthv());
        System.out.println(gameState1.getPlayers());
        System.out.println(gameState1.getMonsters());
        System.out.println(gameState1.getCurrentPlayer());
        System.out.println(gameState1.getWinner());
        System.out.println(gameState1.getLog());



        
        

        
        
        
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    }
    
}

File: PRACTICAS-PDOO/IRRGARTEN/Versiones_en_Java_Ruby/PROYECTO1-RUBY/test_p1.rb
#encoding:utf-8
require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'


class TestP1
  def self.main     #Programa principal
    weapon_1 = Irrgarten::Weapon.new(5.0, 3)
    shield_1 = Irrgarten::Shield.new(3.0, 4)
    game_state_1 = Irrgarten::GameState.new("open", "good", "bored", 2, true, "GOOO")

    puts "Compruebo la clase Weapon"
    puts weapon_1.attack
    puts weapon_1.to_s
    puts weapon_1.discard

    puts "\nCompruebo la clase Shield"
    puts shield_1.protect
    puts shield_1.to_s
    puts weapon_1.discard

    puts "\nCompruebo la clase Dice"
    puts "random_pos: #{Irrgarten::Dice.random_pos(2)}"
    puts "who_starts: #{Irrgarten::Dice.who_starts(5)}"
    puts "random_intelligence: #{Irrgarten::Dice.random_intelligence}"
    puts "random_strength: #{Irrgarten::Dice.random_strength}"
    puts "resurrect_players: #{Irrgarten::Dice.resurrect_player}"
    puts "weapons_reward: #{Irrgarten::Dice.weapons_reward}"
    puts "shields_reward: #{Irrgarten::Dice.shields_reward}"
    puts "health_reward: #{Irrgarten::Dice.health_reward}"
    puts "weapon_power: #{Irrgarten::Dice.weapon_power}"
    puts "shields_power: #{Irrgarten::Dice.shields_power}"
    puts "uses_left: #{Irrgarten::Dice.uses_left}"
    puts "intensity: #{Irrgarten::Dice.intensity(3)}"
    puts "discard_element: #{Irrgarten::Dice.discard_element(5)}"

    puts "\nCompruebo la clase GameState"
    puts game_state_1.labyrinthv
    puts game_state_1.players
    puts game_state_1.monsters
    puts game_state_1.current_player
    puts game_state_1.winner
    puts game_state_1.log
  end
end

TestP1.main